src/components/lmsensors/linux-lmsensors.c

```diff
@@ -399,6 +399,12 @@ link_lmsensors_libraries ()
       dl1 = dlopen(path_name, RTLD_NOW | RTLD_GLOBAL);             // Try to open that path.
    }
 
+   // Step 4: Try another explicit install default.
+   if (dl1 == NULL && lmsensors_root != NULL) {                          // if root given, try it.
+      snprintf(path_name, 1024, "%s/lib/libsensors.so", lmsensors_root);   // PAPI Root check.
```

Comment:
Please use `PATH_MAX` instead of a raw value for string length

Reply:
You can update it in a separate commit

Reply:
Updated as requested with `PATH_MAX`, and added commit to update the same for previous statement.